executor/src/witgen/identity_processor.rs

```diff
@@ -26,23 +24,33 @@ pub struct Machines<'a, 'b, T: FieldElement> {
     machines: Vec<&'b mut KnownMachine<'a, T>>,
 }
 
+enum SplitOption {
+    WithPrevious,
+    WithoutPrevious,
+}
+
 impl<'a, 'b, T: FieldElement> Machines<'a, 'b, T> {
     /// Splits out the machine at `index` and returns it together with a list of all other machines.
     /// As a result, they can be mutated independently.
     /// With #515 implemented (making calling machines immutable), this could be removed as there
     /// would not be a need to split the list of machines.
-    pub fn split<'c>(
+    fn split<'c>(
         &'c mut self,
         index: usize,
+        split_option: SplitOption,
     ) -> (&'c mut KnownMachine<'a, T>, Machines<'a, 'c, T>) {
         let (before, after) = self.machines.split_at_mut(index);
         let (current, after) = after.split_at_mut(1);
         let current: &'c mut KnownMachine<'a, T> = current.first_mut().unwrap();
 
         // Re-borrow machines to convert from `&'c mut &'b mut KnownMachine<'a, T>` to
         // `&'c mut KnownMachine<'a, T>`.
-        let others: Vec<&'c mut KnownMachine<'a, T>> = before
-            .iter_mut()
+        let before_or_none: Box<dyn Iterator<Item = &mut &mut KnownMachine<'a, T>>> =
+            match split_option {
+                SplitOption::WithPrevious => Box::new(before.iter_mut()),
+                SplitOption::WithoutPrevious => Box::new(iter::empty()),
+            };
+        let others: Vec<&'c mut KnownMachine<'a, T>> = before_or_none
```

Comment:
I doubt having the enum argument in this function makes sense in terms of clarity compared to having two separate functions. In either case, it should be documented.

Reply:
Done